packages/fxa-settings/src/components/Settings/Page2faSetup/index.tsx

```diff
@@ -0,0 +1,305 @@
+    if (!totpCode) {
+      showGenericError();
+      goHome();
+      return;
+    }
+    await account.verifyTotp(totpCode);
+  };
+
+  const handleBackupCodeConfirm = async (code: string) => {
+    if (!backupCodes.includes(code)) {
+      setBackupCodeError(
+        ftlMsgResolver.getMsg(
+          'page-2fa-setup-incorrect-backup-code-error',
+          'That code is not correct. Try again.'
+        )
+      );
+      return;
+    }
+
+    try {
+      await account.setRecoveryCodes(backupCodes);
+      await enable2fa();
+    } catch (error) {
+      // for any error other than an incorrect backup code
+      // return to main settings page with generic error message
+      // there is no action the user can take at this step
+      showGenericError();
+      goHome();
+      return;
+    }
+
+    showSuccess();
+    goHome();
+    return;
+  };
+
+  const handleVerifyPhone = async (phoneNumberInput: string) => {
+    const { nationalFormat } = await account.addRecoveryPhone(phoneNumberInput);
+    setPhoneData({
+      phoneNumber: phoneNumberInput,
+      nationalFormat,
+    });
+  };
+
+  const handleResendSms = async () => {
+    await account.addRecoveryPhone(phoneData.phoneNumber);
+  };
+
+  const handleVerifySmsCode = async (code: string) => {
+    // if this errors, error will be handled by try/catch in child component
+    await account.confirmRecoveryPhone(code, phoneData.phoneNumber, true);
+
+    try {
+      await enable2fa();
+    } catch (e) {
+      showGenericError();
+      goHome();
+      return;
+    }
+    await account.refresh('recoveryPhone');
+    // success message controlled by child component
+  };
+
+  return (
+    <VerifiedSessionGuard onDismiss={goHome} onError={goHome}>
+      {currentStep === 1 && (
+        <FlowSetup2faApp
+          verifyCode={handleVerify2faAppCode}
+          onBackButtonClick={goHome}
+          {...{ currentStep, numberOfSteps, localizedPageTitle, totpInfo }}
+        />
+      )}
+
+      {flowHasPhoneChoice && currentStep === 2 && (
+        <FlowSetup2faBackupChoice
+          onSubmitCb={handleBackupChoice}
+          onBackButtonClick={prevStep}
+          {...{ currentStep, numberOfSteps, localizedPageTitle }}
+        />
+      )}
+
+      {((backupMethod === CHOICES.code && currentStep === 3) ||
+        (!flowHasPhoneChoice && currentStep === 2)) && (
+        <FlowSetup2faBackupCodeDownload
+          email={account.email}
+          loading={generatingCodes}
+          onBackButtonClick={prevStep}
+          onContinue={nextStep}
+          {...{ backupCodes, currentStep, numberOfSteps, localizedPageTitle }}
+        />
+      )}
+
+      {((backupMethod === CHOICES.code && currentStep === 4) ||
+        (!flowHasPhoneChoice && currentStep === 3)) && (
+        <FlowSetup2faBackupCodeConfirm
+          currentStep={currentStep}
+          numberOfSteps={numberOfSteps}
+          localizedPageTitle={localizedPageTitle}
```

Comment:
A nit that is probably not worth calling out but if you're going to repush anyways, there's 3 sets of props here in the next component compositions that could use `{...{syntacticSugar}}`